### PR TITLE
Changes/core

### PR DESCRIPTION
**Pull request by cobyge**
Output folder is not set to "Basic", "Advanced", or "{module_name}" depending on what was run.
Removed unused import

### CODE REVIEW

File: IPScanner/Core/helper_functions.py [hunk 1]
```diff
@@ -75,21 +75,28 @@ def validate_args(args, parser):
...
     if not (args.basic or args.advanced or args.module or args.list):
         parser.error("one of the arguments -b/--basic or -a/--advanced or -m/--module is required")
 
+    return
+
 
 def strip_url_protocol(args):
-    if type(args.domain) == str:
+    if args.ip:
+        split_domain = args.ip.split("://")
```
Comment (ofer1katz) [inline IPScanner/Core/helper_functions.py:88]: not a domain

File: IPScanner/Core/helper_functions.py [hunk 2]
```diff
@@ -103,6 +110,7 @@ def validate_ip(ip):
 def create_dir(path):
     if not os.path.exists(path):
         os.mkdir(path)
+    return
```
Comment (ofer1katz) [inline IPScanner/Core/helper_functions.py:113]: redundant

File: IPScanner/Core/helper_functions.py [hunk 3]
```diff
@@ -75,21 +75,28 @@ def validate_args(args, parser):
...
+    if args.domain:
         split_domain = args.domain.split("://")
         if len(split_domain) > 1:
             args.url_scheme = split_domain[0]
         args.domain = split_domain[-1]
-        return
-    else:
-        return
+
+    return
```
Comment (ofer1katz) [inline IPScanner/Core/helper_functions.py:99]: redundant

File: IPScanner/Core/helper_functions.py [hunk 4]
```diff
@@ -75,21 +75,28 @@ def validate_args(args, parser):
     if args.domain:
         domain_ip = resolve_domain(args.domain)
         parser.error("could not resolve argument -d/--domain to ip") if not domain_ip else None
-        args.ip = domain_ip or args.ip
+        args.ip = args.ip or domain_ip
 
     if not (args.basic or args.advanced or args.module or args.list):
         parser.error("one of the arguments -b/--basic or -a/--advanced or -m/--module is required")
 
+    return
```
Comment (ofer1katz) [inline IPScanner/Core/helper_functions.py:83]: redundant

File: IPScanner/Core/helper_functions.py [hunk 5]
```diff
@@ -22,7 +21,8 @@ def print_banner():
     colors = [codes[color] for color in codes if color not in bad_colors]
     random_foreground = random.choice(colors)
     print(random_foreground + consts.BANNER)
-    print(colorama.Fore.RESET + colorama.Back.RESET),
+    print(colorama.Fore.RESET + colorama.Back.RESET)
+    return
```
Comment (ofer1katz) [inline IPScanner/Core/helper_functions.py:26]: redundant

File: IPScanner/Core/scanner.py [hunk 6]
```diff
@@ -18,6 +18,7 @@ def __init__(self, args, module_dict):
         colorama.init()
         self.args = args
         self.module_dict = module_dict
+        return
```
Comment (ofer1katz) [inline IPScanner/Core/scanner.py:21]: redundant

File: IPScanner/Core/scanner.py [hunk 7]
```diff
@@ -74,23 +74,31 @@ def __init__(self, args, module_dict):
...
+            return
+
+        result_dir = consts.SCAN_MODE_DICT.get(module_type, self.args.module)
+        if not result_dir:
+            print(colorama.Fore.RED + "[-] Error with getting path to save, will be saved to Output instead of subdirectory")
+            print(colorama.Fore.RESET + colorama.Back.RESET)
+            result_dir = ""
+
+        self.save_scan_results(result_dir, results)
+        return
```
Comment (ofer1katz) [inline IPScanner/Core/scanner.py:86]: redundant

File: IPScanner/Core/scanner.py [hunk 8]
```diff
@@ -74,23 +74,31 @@ def __init__(self, args, module_dict):
...
-        path = os.path.join(consts.PRODUCTS_DIR_PATH, scan_mode, r'{}.txt'.format(filename))
+        path = os.path.join(consts.PRODUCTS_DIR_PATH, result_dir, r'{}.txt'.format(filename))
 
         print(colorama.Fore.GREEN + "[+] Writing results to: {}".format(path))
         print(colorama.Fore.RESET + colorama.Back.RESET)
 
         with open(path, 'a') as products_file:
             json.dump(results, products_file, ensure_ascii=False, indent=4)
             products_file.write("\n\n")
+        return
```
Comment (ofer1katz) [inline IPScanner/Core/scanner.py:101]: redundant

File: IPScanner/Core/scanner.py [hunk 9]
```diff
@@ -26,10 +27,9 @@ def __init__(self, args, module_dict):
     async def scan(self):
         """Basic function that is responsible for all scans""" 
 
-        if self.args.basic:
-            selected_modules = {module: self.module_dict[module] for module in self.module_dict if self.module_dict[module]['module_type'] == 'Basic'}
-        elif self.args.advanced:
-            selected_modules = {module: self.module_dict[module] for module in self.module_dict if self.module_dict[module]['module_type'] == 'Advanced'}
+        module_type = self.args.basic or self.args.advanced or None
```
Comment (ofer1katz) [inline IPScanner/Core/scanner.py:30]: You can use the dest argument for the function add_argument and store both basic and advanced consts in module_type

File: IPScanner/Core/helper_functions.py (hunk 2 above)
Comment (cobyge) [reply IPScanner/Core/helper_functions.py:113]: PEP8 States that we need a return at the end of every function

File: IPScanner/Core/helper_functions.py (hunk 3 above)
Comment (cobyge) [reply IPScanner/Core/helper_functions.py:99]: PEP8 States that we need a return at the end of every function

File: IPScanner/Core/helper_functions.py (hunk 4 above)
Comment (cobyge) [reply IPScanner/Core/helper_functions.py:83]: PEP8 States that we need a return at the end of every function

File: IPScanner/Core/helper_functions.py (hunk 5 above)
Comment (cobyge) [reply IPScanner/Core/helper_functions.py:26]: PEP8 States that we need a return at the end of every function

File: IPScanner/Core/scanner.py (hunk 6 above)
Comment (cobyge) [reply IPScanner/Core/scanner.py:21]: PEP8 States that we need a return at the end of every function

File: IPScanner/Core/scanner.py (hunk 7 above)
Comment (cobyge) [reply IPScanner/Core/scanner.py:86]: PEP8 States that we need a return at the end of every function

File: IPScanner/Core/scanner.py (hunk 8 above)
Comment (cobyge) [reply IPScanner/Core/scanner.py:101]: PEP8 States that we need a return at the end of every function

File: IPScanner/Core/helper_functions.py (hunk 1 above)
Comment (cobyge) [reply IPScanner/Core/helper_functions.py:88]: Fixed

File: IPScanner/Core/scanner.py (hunk 9 above)
Comment (cobyge) [reply IPScanner/Core/scanner.py]: Fixed

File: IPScanner/Core/helper_functions.py [hunk 10]
```diff
@@ -21,7 +21,8 @@ def print_banner():
     colors = [codes[color] for color in codes if color not in bad_colors]
     random_foreground = random.choice(colors)
     print(random_foreground + consts.BANNER)
-    print(colorama.Fore.RESET + colorama.Back.RESET),
+    print(colorama.Fore.RESET + colorama.Back.RESET)
```
[the remark this answers is not in the document]
Comment (cobyge) [reply IPScanner/Core/helper_functions.py:24]: PEP8 States that we need a return at the end of every function

**Pull request merged**